### ISSUE TITLE
False positive ``unused-import`` for type annotations inside quotes

### ISSUE THREAD

**Issue by udifuchs** [PyCQA/pylint#3765]
pylint complains about `W0611: Unused List imported from typing (unused-import)` for this code:
```python
from typing import List
a: "List[int]" = []

class B:
    def get_all_classes(self) -> "List[B]":
        pass
```
The first two lines already demonstrate the problem, but in the definition of `a` I could easily remove the quotes to fix the issue.

In the definition of `B.get_all_classes` I cannot remove the quotes, because that would cause python to complain:  `NameError: name 'B' is not defined`.

Removing the import would cause mypy to complain `Name 'List' is not defined`.

It seems that pylint does not support type annotations inside quotes.

### pylint --version output
pylint 2.5.3
astroid 2.4.2
Python 3.8.2 (default, Jul 16 2020, 14:00:26) 
[GCC 9.3.0]

**Comment by BurningLights** [comment]
Yes, I see the same thing with pylint. I have the issue with typing Django QuerySets. Doing it without the quotes produces the python error "TypeError: 'type' object is not subscriptable." So, I must do it with the quotes.

**pylint --version output**
pylint 2.5.3
astroid 2.4.2
Python 3.6.8 (default, Jan 14 2019, 11:02:34) 
[GCC 8.0.1 20180414 (experimental) [trunk revision 259383]]

**Comment by mbyrnepr2** [comment]
Looks like this one has been fixed along the way - it looks like its thanks to [this logic](https://github.com/PyCQA/pylint/blob/main/pylint/checkers/variables.py#L2830-L2832).
